internal/nodes/avi_model_evh_nodes.go

```diff
@@ -1624,8 +1624,13 @@ func buildWithInfraSettingForEvh(key string, vs *AviEvhVsNode, vsvip *AviVSVIPNo
 				vsvip.VipNetworks = vipNetworks
 			}
 		}
+		enablePublicIP := false
+		if infraSetting.Spec.Network.EnablePublicIP != nil && lib.IsPublicCloud() {
```

Comment:
Can you add a cloud check for this ? public ip can only be enabled with public clouds and Openstack